switchsdk/src/androidTest/java/net/gini/switchsdk/uitests/ReviewPictureScreenTest.java

```diff
@@ -0,0 +1,162 @@
+package net.gini.switchsdk.uitests;
+
+import static android.support.test.InstrumentationRegistry.getInstrumentation;
+import static android.support.test.espresso.Espresso.onView;
+import static android.support.test.espresso.Espresso.openActionBarOverflowOrOptionsMenu;
+import static android.support.test.espresso.action.ViewActions.click;
+import static android.support.test.espresso.assertion.ViewAssertions.matches;
+import static android.support.test.espresso.matcher.RootMatchers.isDialog;
+import static android.support.test.espresso.matcher.ViewMatchers.isDisplayed;
+import static android.support.test.espresso.matcher.ViewMatchers.withId;
+import static android.support.test.espresso.matcher.ViewMatchers.withText;
+
+import static net.gini.switchsdk.uitests.ScreenUtils.childAtPosition;
+
+import static org.hamcrest.Matchers.allOf;
+
+import android.Manifest;
+import android.content.Context;
+import android.content.Intent;
+import android.content.SharedPreferences;
+import android.os.SystemClock;
+import android.support.test.InstrumentationRegistry;
+import android.support.test.espresso.ViewInteraction;
+import android.support.test.filters.LargeTest;
+import android.support.test.rule.ActivityTestRule;
+import android.support.test.rule.GrantPermissionRule;
+import android.support.test.runner.AndroidJUnit4;
+import android.view.View;
+
+import net.gini.switchsdk.OnboardingManager;
+import net.gini.switchsdk.R;
+import net.gini.switchsdk.SwitchSdk;
+
+import org.hamcrest.core.IsInstanceOf;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+@RunWith(AndroidJUnit4.class)
+@LargeTest
+public class ReviewPictureScreenTest {
+    private static final int TIME_TO_WAIT_BETWEEN_STEPS = 1000;
+    @Rule
+    public ActivityTestRule<HostActivity> mActivityTestRule = new ActivityTestRule
+            <HostActivity>(HostActivity.class) {
+        @Override
+        protected Intent getActivityIntent() {
+            Context context = InstrumentationRegistry.getInstrumentation().getTargetContext();
+            SwitchSdk switchSdk = SwitchSdk.init(context, "", "", "");
+            Intent intent = new Intent(context, HostActivity.class);
+
+            intent.putExtras(switchSdk.getSwitchSdkIntent());
+            return intent;
+        }
+    };
+    @Rule
+    public GrantPermissionRule mRuntimePermissionRule = GrantPermissionRule.grant(
+            Manifest.permission.CAMERA);
+
+    @Test
+    public void menu_cancelShouldDisplayDialog() {
+
+        openActionBarOverflowOrOptionsMenu(getInstrumentation().getTargetContext());
+
+        SystemClock.sleep(TIME_TO_WAIT_BETWEEN_STEPS);
+        ViewInteraction appCompatTextView = onView(
+                allOf(withId(R.id.title), withText(R.string.menu_entry_cancel), isDisplayed()));
+        appCompatTextView.perform(click());
+
+        SystemClock.sleep(TIME_TO_WAIT_BETWEEN_STEPS);
```

Comment:
having flaiky UI test maybe. It just to be sure that the view is there when we test